uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChangeTest.php

```diff
@@ -443,6 +443,112 @@ public function testLeaveRequestBalanceForEntitlementCanSumBalanceChangesCreated
     $this->assertEquals(-1, $balance);
   }
 
+  public function testLeaveRequestBalanceForEntitlementCanExcludeBalanceChangesForPublicHolidayLeaveRequests() {
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+    $entitlement = $this->createLeavePeriodEntitlementMockForBalanceTests(
+      new DateTime('today'),
+      new DateTime('+100 days')
+    );
+
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement);
+    $this->assertEquals(0, $balance);
+
+    // This will deduct 11 days
+    $this->createLeaveRequestBalanceChange(
+      $entitlement->type_id,
+      $entitlement->contact_id,
+      $leaveRequestStatuses['Approved'],
+      date('Y-m-d'),
+      date('Y-m-d', strtotime('+10 days'))
+    );
+
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = date('Y-m-d', strtotime('+30 days'));
+    PublicHolidayLeaveRequestFabricator::fabricate($entitlement->contact_id, $publicHoliday);
+
+    // Balance excluding the days deducted from the leave request
+    $excludePublicHolidays = true;
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement, [], null, null, $excludePublicHolidays);
+    $this->assertEquals(-11, $balance);
+  }
+
+  public function testLeaveRequestBalanceForEntitlementCanIncludeOnlyBalanceChangesForPublicHolidayLeaveRequests() {
+    $leaveRequestStatuses = array_flip(LeaveRequest::buildOptions('status_id'));
+    $entitlement = $this->createLeavePeriodEntitlementMockForBalanceTests(
+      new DateTime('today'),
+      new DateTime('+100 days')
+    );
+
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement($entitlement);
+    $this->assertEquals(0, $balance);
+
+    // This will deduct 11 days
+    $this->createLeaveRequestBalanceChange(
+      $entitlement->type_id,
+      $entitlement->contact_id,
+      $leaveRequestStatuses['Approved'],
+      date('Y-m-d'),
+      date('Y-m-d', strtotime('+10 days'))
+    );
+
+    $publicHoliday1 = new PublicHoliday();
+    $publicHoliday1->date = date('Y-m-d', strtotime('+30 days'));
+    PublicHolidayLeaveRequestFabricator::fabricate($entitlement->contact_id, $publicHoliday1);
+
+    $publicHoliday2 = new PublicHoliday();
+    $publicHoliday2->date = date('Y-m-d', strtotime('+47 days'));
+    PublicHolidayLeaveRequestFabricator::fabricate($entitlement->contact_id, $publicHoliday2);
+
+    // Balance excluding the days deducted from the leave request
+    $includePublicHolidaysOnly = true;
+    $balance = LeaveBalanceChange::getLeaveRequestBalanceForEntitlement(
+      $entitlement,
+      [],
+      null,
+      null,
+      false,
+      $includePublicHolidaysOnly
+    );
+    $this->assertEquals(-2, $balance);
+  }
+
+  public function testLeaveRequestBalanceForEntitlementWithParamsToBothExcludeAndIncludePublicHolidaysShouldReturnZero() {
```

Comment:
It's because we're both excluding and including public holidays, which will result a q condition something like this:
```sql
WHERE leave_balance_change.type_id <> 'Public Holiday' AND leave_balance_change.type_id = 'Public Holiday'
```

There's no way for a balance change to be both a public holiday and not a public holiday, so nothing will be returned and the sum will be 0.